mailbox/store/src/test/java/org/apache/james/mailbox/store/StoreRightManagerTest.java

```diff
@@ -24,17 +24,176 @@
+
+    @Before
+    public void setup() throws MailboxException {
+        alice = "Alice";
+        aliceSession = new MockMailboxSession(alice);
+        MailboxSessionMapperFactory mockedMapperFactory = mock(MailboxSessionMapperFactory.class);
+        mockedMailboxMapper = mock(MailboxMapper.class);
+        mailboxAclResolver = new UnionMailboxACLResolver();
+        groupMembershipResolver = new SimpleGroupMembershipResolver();
+        when(mockedMapperFactory.getMailboxMapper(aliceSession))
+            .thenReturn(mockedMailboxMapper);
+        storeRightManager = new StoreRightManager(mockedMapperFactory,
+                                                  mailboxAclResolver,
+                                                  groupMembershipResolver);
+    }
+
+    @Test
+    public void hasRighShouldThrowMailboxNotFoundExceptionWhenMailboxDoesNotExist() throws MailboxException {
+        expectedException.expect(MailboxNotFoundException.class);
+
+        MailboxPath mailboxPath = MailboxPath.forUser(alice, "unexisting mailbox");
+        when(mockedMailboxMapper.findMailboxByPath(mailboxPath)).thenThrow(MailboxNotFoundException.class);
+        storeRightManager.hasRight(mailboxPath, Right.Read, aliceSession);
+    }
+
+    @Test
+    public void hasRighShouldReturnTrueWhenTheUserOwnTheMailbox() throws MailboxException {
+        Mailbox mailbox = mock(Mailbox.class);
+        when(mailbox.getUser()).thenReturn(alice);
+
+        assertThat(storeRightManager.hasRight(mailbox, Right.Write, aliceSession))
+            .isFalse();
+    }
+
+    @Test
+    public void hasRighShouldReturnTrueWhenTheUserDoesnotOwnTheMailboxButHaveTheCorrectedRightOnIt() throws MailboxException {
+        Mailbox mailbox = mock(Mailbox.class);
+        when(mailbox.getUser()).thenReturn("bob");
+        when(mailbox.getACL()).thenReturn(new MailboxACL(new MailboxACL.Entry(alice, Right.Write)));
+
+        assertThat(storeRightManager.hasRight(mailbox, Right.Write, aliceSession))
+            .isTrue();
+    }
+
+    @Test
+    public void hasRighShouldReturnTrueWhenTheUserDoesnotOwnTheMailboxButHaveHasLeastTheCorrectedRightOnIt() throws MailboxException {
+        Mailbox mailbox = mock(Mailbox.class);
+        when(mailbox.getUser()).thenReturn("bob");
+        when(mailbox.getACL()).thenReturn(new MailboxACL(new MailboxACL.Entry(alice, Right.Write, Right.Lookup)));
+
+        assertThat(storeRightManager.hasRight(mailbox, Right.Write, aliceSession))
+            .isTrue();
+    }
+
+    @Test
+    public void hasRighShouldReturnFalseWhenTheUserDoesNotOwnTheMailboxAndAsNoRightOnIt() throws MailboxException {
+        Mailbox mailbox = mock(Mailbox.class);
+        when(mailbox.getUser()).thenReturn("bob");
+
+        assertThat(storeRightManager.hasRight(mailbox, Right.Write, aliceSession))
+            .isFalse();
+    }
+
+    @Test
+    public void isReadWriteShouldReturnTrueWhenUserHasInsertRightOnMailbox() throws Exception {
```

Comment:
is it normal? I mean someone could have insert right (right to add things) but not read right (right to read things), no?

Reply:
I agree with you, but this behavior is in the implementation since a long time;
this test is just extending the coverage.

I don't know if we have to change that or no, we have to check the impacts first.

Do you agree to create a new issue for that?

Reply:
yes ok, let's write an issue about this (strange) function